vkcubepp (KHR_display) fails to build since v1.3.209

Regressed by c2f1a9494a90

```c++
$ cmake -DCUBE_WSI_SELECTION=DISPLAY -B /tmp/vulkan-tools_build
$ cmake --build /tmp/vulkan-tools_build
[...]
FAILED: cube/CMakeFiles/vkcubepp.dir/cube.cpp.o
/usr/bin/c++ -DVK_ENABLE_BETA_EXTENSIONS -DVK_USE_PLATFORM_DISPLAY_KHR -I/tmp/vulkan-tools_build/cube -Icube -I/tmp/vulkan-tools_build/cube/.. -isystem /usr/local/include -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -fno-strict-aliasing -fno-builtin-memcmp -Werror -std=c++11 -fno-rtti -fvisibility=hidden -MD -MT cube/CMakeFiles/vkcubepp.dir/cube.cpp.o -MF cube/CMakeFiles/vkcubepp.dir/cube.cpp.o.d -o cube/CMakeFiles/vkcubepp.dir/cube.cpp.o -c cube/cube.cpp
cube/cube.cpp:2923:13: error: use of undeclared identifier 'plane_props'
        if (plane_props[plane_index].currentDisplay && (plane_props[plane_index].currentDisplay != display)) {
            ^
cube/cube.cpp:2923:57: error: use of undeclared identifier 'plane_props'
        if (plane_props[plane_index].currentDisplay && (plane_props[plane_index].currentDisplay != display)) {
                                                        ^
cube/cube.cpp:2952:84: error: use of undeclared identifier 'mode_props'
    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(mode_props.displayMode, plane_index);
                                                                                   ^
cube/cube.cpp:2952:108: error: use of undeclared identifier 'plane_index'
    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(mode_props.displayMode, plane_index);
                                                                                                           ^
cube/cube.cpp:2969:27: error: use of undeclared identifier 'mode_props'
    image_extent.setWidth(mode_props.parameters.visibleRegion.width).setHeight(mode_props.parameters.visibleRegion.height);
                          ^
cube/cube.cpp:2969:80: error: use of undeclared identifier 'mode_props'
    image_extent.setWidth(mode_props.parameters.visibleRegion.width).setHeight(mode_props.parameters.visibleRegion.height);
                                                                               ^
cube/cube.cpp:2974:53: error: use of undeclared identifier 'plane_props'
                                .setPlaneStackIndex(plane_props[plane_index].currentStackIndex)
                                                    ^
cube/cube.cpp:2974:65: error: use of undeclared identifier 'plane_index'; did you mean 'frame_index'?
                                .setPlaneStackIndex(plane_props[plane_index].currentStackIndex)
                                                                ^~~~~~~~~~~
                                                                frame_index
cube/cube.cpp:363:14: note: 'frame_index' declared here
    uint32_t frame_index = 0;
             ^
cube/cube.cpp:2972:49: error: use of undeclared identifier 'mode_props'
                                .setDisplayMode(mode_props.displayMode)
                                                ^
cube/cube.cpp:2973:48: error: use of undeclared identifier 'plane_index'
                                .setPlaneIndex(plane_index)
                                               ^
10 errors generated.
```

same here:

```
BUILD      vulkan-tools (target)
    TOOLCHAIN      cmake (auto-detect)
Executing (target): cmake -GNinja -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_TOOLCHAIN_FILE=/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/etc/cmake-armv8a-libreelec-linux-gnueabihf.conf -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DBUILD_VULKANINFO=ON -DBUILD_ICD=OFF -DINSTALL_ICD=OFF -DBUILD_WSI_DIRECTFB_SUPPORT=OFF -Wno-dev -DBUILD_CUBE=ON -DBUILD_WSI_XCB_SUPPORT=OFF -DBUILD_WSI_XLIB_SUPPORT=OFF -DBUILD_WSI_WAYLAND_SUPPORT=OFF -DCUBE_WSI_SELECTION=DISPLAY /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using find_package to locate Vulkan
-- Found Vulkan: /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libvulkan.so  
-- Found VulkanHeaders: /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include  
-- Found VulkanRegistry: /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/share/vulkan/registry  
-- Vulkan FOUND = TRUE
-- Vulkan Lib Dir = /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib
-- Vulkan Lib = /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libvulkan.so
-- Vulkan Headers Include = /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include
-- Vulkan Headers Registry = /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/share/vulkan/registry
-- Looking for glslangValidator
-- Looking for glslangValidator - /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/bin/glslangValidator
-- Found PkgConfig: /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/bin/pkg-config (found version "0.29.2") 
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Configuring done
-- Generating done
-- Build files have been written to: /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/.armv8a-libreelec-linux-gnueabihf
Executing (target): ninja 
[1/8] Compiling cube vertex shader
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.vert
[2/8] Compiling cube fragment shader
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.frag
[3/8] Building C object cube/CMakeFiles/vkcube.dir/cube.c.o
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.c: In function 'debug_messenger_callback':
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.c:529:25: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  529 |                         (void *)(pCallbackData->pObjects[object].objectHandle), pCallbackData->pObjects[object].pObjectName);
      |                         ^
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.c:533:25: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  533 |                         (void *)(pCallbackData->pObjects[object].objectHandle));
      |                         ^
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.c: In function 'demo_draw_build_cmd':
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.c:766:29: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  766 |             .objectHandle = (uint64_t)cmd_buf,
      |                             ^
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.c: In function 'find_display_gpu':
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.c:3119:14: warning: variable 'result' set but not used [-Wunused-but-set-variable]
 3119 |     VkResult result;
      |              ^~~~~~
[4/8] Linking C executable cube/vkcube
[5/8] Building CXX object cube/CMakeFiles/vkcubepp.dir/cube.cpp.o
FAILED: cube/CMakeFiles/vkcubepp.dir/cube.cpp.o 
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-g++ -DVK_ENABLE_BETA_EXTENSIONS -DVK_USE_PLATFORM_DISPLAY_KHR -I/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/.armv8a-libreelec-linux-gnueabihf/cube -I/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube -I/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/.armv8a-libreelec-linux-gnueabihf/cube/.. -march=armv8-a+crc -mtune=cortex-a73.cortex-a53 -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -mfloat-abi=hard -mfpu=neon-fp-armv8 -Wall -pipe  -O2 -fomit-frame-pointer -DNDEBUG -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -fno-strict-aliasing -fno-builtin-memcmp -Wimplicit-fallthrough=0 -std=c++11 -fno-rtti -fvisibility=hidden -O3 -DNDEBUG -MD -MT cube/CMakeFiles/vkcubepp.dir/cube.cpp.o -MF cube/CMakeFiles/vkcubepp.dir/cube.cpp.o.d -o cube/CMakeFiles/vkcubepp.dir/cube.cpp.o -c /mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.cpp
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.cpp: In member function 'vk::Result Demo::create_display_surface()':
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.cpp:2923:13: error: 'plane_props' was not declared in this scope
 2923 |         if (plane_props[plane_index].currentDisplay && (plane_props[plane_index].currentDisplay != display)) {
      |             ^~~~~~~~~~~
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.cpp:2952:84: error: 'mode_props' was not declared in this scope
 2952 |     vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(mode_props.displayMode, plane_index);
      |                                                                                    ^~~~~~~~~~
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.cpp:2952:108: error: 'plane_index' was not declared in this scope; did you mean 'frame_index'?
 2952 |     vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(mode_props.displayMode, plane_index);
      |                                                                                                            ^~~~~~~~~~~
      |                                                                                                            frame_index
/mnt/dev/LibreELEC-RR/build.LibreELEC-AMLG12B.arm-11.0-devel/build/vulkan-tools-1.3.209/cube/cube.cpp:2974:53: error: 'plane_props' was not declared in this scope; did you mean 'planeCaps'?
 2974 |                                 .setPlaneStackIndex(plane_props[plane_index].currentStackIndex)
      |                                                     ^~~~~~~~~~~
      |                                                     planeCaps
ninja: build stopped: subcommand failed.
```

--- a/cube/cube.cpp
+++ b/cube/cube.cpp
@@ -994,11 +994,13 @@ int find_display_gpu(int gpu_number, const std::vector<vk::PhysicalDevice> &phys
     if (gpu_number >= 0) {
         auto display_props_return = physical_devices[gpu_number].getDisplayPropertiesKHR();
         VERIFY(display_props_return.result == vk::Result::eSuccess);
+        display_count = display_props_return.value.size();
     } else {
         for (uint32_t i = 0; i < physical_devices.size(); i++) {
             auto display_props_return = physical_devices[i].getDisplayPropertiesKHR();
             VERIFY(display_props_return.result == vk::Result::eSuccess);
             if (display_props_return.value.size() > 0) {
+                display_count = display_props_return.value.size();
                 gpu_return = i;
                 break;
             }
@@ -2915,12 +2917,14 @@ vk::Result Demo::create_display_surface() {
         fflush(stdout);
         exit(1);
     }
+    auto display_plane_props = display_plane_props_return.value;
 
     vk::Bool32 found_plane = VK_FALSE;
+    uint32_t plane_found = 0;
     // Find a plane compatible with the display
-    for (uint32_t plane_index = 0; plane_index < display_plane_props_return.value.size(); plane_index++) {
+    for (uint32_t plane_index = 0; plane_index < display_plane_props.size(); plane_index++) {
         // Disqualify planes that are bound to a different display
-        if (plane_props[plane_index].currentDisplay && (plane_props[plane_index].currentDisplay != display)) {
+        if (display_plane_props[plane_index].currentDisplay && (display_plane_props[plane_index].currentDisplay != display)) {
             continue;
         }
 
@@ -2934,6 +2938,7 @@ vk::Result Demo::create_display_surface() {
         for (const auto &supported_display : display_plane_supported_displays_return.value) {
             if (supported_display == display) {
                 found_plane = VK_TRUE;
+                plane_found = plane_index;
                 break;
             }
         }
@@ -2949,7 +2954,7 @@ vk::Result Demo::create_display_surface() {
         exit(1);
     }
 
-    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(mode_props.displayMode, plane_index);
+    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(display_mode_prop.displayMode, plane_found);
     // Find a supported alpha mode
     vk::DisplayPlaneAlphaFlagBitsKHR alphaMode = vk::DisplayPlaneAlphaFlagBitsKHR::eOpaque;
     std::array<vk::DisplayPlaneAlphaFlagBitsKHR, 4> alphaModes = {
@@ -2966,12 +2971,13 @@ vk::Result Demo::create_display_surface() {
     }
 
     vk::Extent2D image_extent{};
-    image_extent.setWidth(mode_props.parameters.visibleRegion.width).setHeight(mode_props.parameters.visibleRegion.height);
+    image_extent.setWidth(display_mode_prop.parameters.visibleRegion.width)
+        .setHeight(display_mode_prop.parameters.visibleRegion.height);
 
     auto const createInfo = vk::DisplaySurfaceCreateInfoKHR()
-                                .setDisplayMode(mode_props.displayMode)
-                                .setPlaneIndex(plane_index)
-                                .setPlaneStackIndex(plane_props[plane_index].currentStackIndex)
+                                .setDisplayMode(display_mode_prop.displayMode)
+                                .setPlaneIndex(plane_found)
+                                .setPlaneStackIndex(display_plane_props[plane_found].currentStackIndex)
                                 .setGlobalAlpha(1.0f)
                                 .setAlphaMode(alphaMode)
                                 .setImageExtent(image_extent);

